### ISSUE TITLE
Compose multiplatform iOS error: A Koin Application has already been started

### ISSUE THREAD

**Issue by SnehaLizPaul** [InsertKoinIO/koin#1636]
I am using koin for compose multiplatform (android and iOS). Android app running perfectly fine. When I run iOS app getting an error 

> Uncaught Kotlin exception: org.koin.core.error.KoinAppAlreadyStartedException: A Koin Application has already been started

My main.ios.kt looks like:
```
fun MainViewController() = ComposeUIViewController {
    initKoin()
    Navigator(SplashScreen())
}
```

Android MainActivity looks like: 
```
override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    initKoin {
      androidContext(applicationContext)
    }

    setContent{
      Navigator(SplashScreen())
    }
}
```

Shared dependency looks like:
```
val commonMain by getting {
    resources.srcDirs("resources")
    dependencies {
     implementation("io.insert-koin:koin-core:3.3.3")
    }
}
```

init koin in shared module looks like:
```
fun initKoin(appDeclaration: KoinAppDeclaration = {}) {
  startKoin {
    appDeclaration()
    modules(userModule)
  }
}
```

**Comment by hoc081098** [comment]
Call `initKoin()` in your [UIApplicationDelegate](https://developer.apple.com/documentation/uikit/uiapplicationdelegate) or SwiftUI App.
See https://github.com/hoc081098/GithubSearchKMM/blob/313a8b3bb7aa976518a415daf2864efd0663cac4/iosApp/iosApp/iOSApp.swift#L7, https://github.com/hoc081098/GithubSearchKMM/blob/313a8b3bb7aa976518a415daf2864efd0663cac4/shared/src/iosMain/kotlin/com/hoc081098/github_search_kmm/koin.kt#L37

**Comment by hoc081098** [comment]
@SnehaLizPaul make sure that you call initKoin only once

_Sent from my 2201117TG using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by hoc081098** [comment]
Remove initKoin 
```diff
fun MainViewController() = ComposeUIViewController { 
-   initKoin() 
   Navigator(MainScreen()) 
 }
```

Call initKoin() in 
```swift
struct App {
  init { initKoin() }
}

**Comment by SnehaLizPaul** [comment]
```
init() {
        AppModuleKt.doInitKoin()
    }
```

Fixed the issue. Thanks you.

**Comment by blonsky95** [comment]
Hi @SnehaLizPaul I ran into this same error and your fix did the job, however I run into a different error, I started a new issue here https://github.com/InsertKoinIO/koin/issues/1647 but did you come across it? (In your case it wouldn't find AppModuleKt in scope)

**Comment by mustfaibra** [comment]
> ```
> init() {
>         AppModuleKt.doInitKoin()
>     }
> ```
> 
> Fixed the issue. Thanks you.

It's strange, but iOS will not find this method if it's not starting with do, but it worked!

**Comment by hoc081098** [comment]
> > ```
> > init() {
> >         AppModuleKt.doInitKoin()
> >     }
> > ```
> > 
> > Fixed the issue. Thanks you.
> 
> It's strange, but iOS will not find this method if it's not starting with do, but it worked!

It requires compiling Kotlin first

**Comment by mustfaibra** [comment]
> > > ```
> > > init() {
> > >         AppModuleKt.doInitKoin()
> > >     }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Fixed the issue. Thanks you.
> > 
> > 
> > It's strange, but iOS will not find this method if it's not starting with do, but it worked!
> 
> It requires compiling Kotlin first

What do you mean? without the prefix "do" the compiler will not start compiling the Kotlin code?

**Comment by hoc081098** [comment]
> 
> 
> > > > ```
> > > > init() {
> > > >         AppModuleKt.doInitKoin()
> > > >     }
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > Fixed the issue. Thanks you.
> > > 
> > > 
> > > It's strange, but iOS will not find this method if it's not starting with do, but it worked!
> > 
> > It requires compiling Kotlin first
> 
> What do you mean? without the prefix "do" the compiler will not start compiling the Kotlin code?

Ah, Kotlin methods called in Swift/Object starts with `do` prefix.
You can check in ObjectiveC/Swift interop docs https://kotlinlang.org/docs/native-objc-interop.html

**Comment by Ashkan-san** [comment]
For me the issue was that I called `initKoin()` in the MainViewController and in the iosApp as well. Removed it from the MainViewController and added `ModuleKt.doInitKoin(appDeclaration: {_ in})` in the init part of the iosApp. ModuleKt should be the file name of where your initKoin method is in.
If you can't find the declaration you have to add `import shared` or whatever your shared module name is to the swift code.

**Comment by rogribas** [comment]
In my case I fixed by only initializing it on the App.kt on commonMain with a boolean to prevent retriggering by recomposition:

```
private var isKoinInitialized = false

@Composable
@Preview
fun App() {
    if (!isKoinInitialized) {
        initializeKoin()
        isKoinInitialized = true
    }
    ...
}

fun initializeKoin() {
    startKoin {
        modules(realmModule)
    }
}
```
